manifests/cli.pp

```diff
@@ -0,0 +1,36 @@
+class puppetdb::cli (
+  $confdir       = '/etc/puppetlabs/client-tools',
+  $puppetdb_urls = ['http://127.0.0.1:8080'],
+  $version       = 'present',
+  $ssl_cert_path = undef,
+  $ssl_key_path  = undef,
+  $localcacert   = undef,
+) inherits puppetdb::params {
+
+  if !($puppetdb::params::puppetdb_version in ['present','absent'])
+  and versioncmp($puppetdb::params::puppetdb_version, '4.0.0') >= 0 {
+    err('puppet-client-tools is only compatible with PuppetDB 4.0.0 and greater')
+  }
+
+  $puppetdb_cli_config = "${confdir}/puppetdb.conf"
+  $_puppetdb_cli_config = { 'puppetdb' =>  {
+    'server_urls' => $puppetdb_urls,
+    'cacert'      => $localcacert,
+    'cert'        => $ssl_cert_path,
+    'key'         => $ssl_key_path,
+    }
+  }
+
+  package { 'puppet-client-tools':
+    ensure => $version,
+  }
+
+  file { $puppetdb_cli_config:
+    ensure  => 'present',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0444',
+    content => inline_template('<%= @_puppetdb_cli_config.to_json %>'),
+    require => Package['puppet-client-tools'],
+  }
```

Comment:
All this code is mostly lifted from the PE module (puppetlabs-puppet_enterprise).